src/main/java/com/autotune/analyzer/services/Summarize.java

```diff
@@ -0,0 +1,915 @@
+import com.autotune.analyzer.utils.AnalyzerErrorConstants;
+import com.autotune.analyzer.utils.GsonUTCDateAdapter;
+import com.autotune.database.service.ExperimentDBService;
+import com.autotune.utils.KruizeConstants;
+import com.autotune.utils.KruizeSupportedTypes;
+import com.autotune.utils.MetricsConfig;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.micrometer.core.instrument.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.CHARACTER_ENCODING;
+import static com.autotune.analyzer.utils.AnalyzerConstants.ServiceConstants.JSON_CONTENT_TYPE;
+
+public class Summarize extends HttpServlet {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Summarize.class);
+    HashMap<String, SummarizeAPIObject> clusterSummaryCacheMap = new HashMap<>();
+    HashMap<String, SummarizeAPIObject> namespaceSummaryCacheMap = new HashMap<>();
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Timer.Sample timerListRec = Timer.start(MetricsConfig.meterRegistry());
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.setContentType(JSON_CONTENT_TYPE);
+        response.setCharacterEncoding(CHARACTER_ENCODING);
+        List<SummarizeAPIObject> summarizeAPIObjectList = new ArrayList<>();
+        String summarizeType = request.getParameter(KruizeConstants.JSONKeys.SUMMARIZE_TYPE);
+        String clusterName = request.getParameter(KruizeConstants.JSONKeys.CLUSTER_NAME);
+        String namespaceName = request.getParameter(KruizeConstants.JSONKeys.NAMESPACE_NAME);
+        String fetchFromDB = request.getParameter(KruizeConstants.JSONKeys.FETCH_FROM_DB);
+
+        // validate Query params
+        Set<String> invalidParams = new HashSet<>();
+        for (String param : request.getParameterMap().keySet()) {
+            if (!KruizeSupportedTypes.SUMMARIZE_PARAMS_SUPPORTED.contains(param)) {
+                invalidParams.add(param);
+            }
+        }
+        if (invalidParams.isEmpty()) {
+            // Set default value if absent
+            if (summarizeType == null || summarizeType.isEmpty())
+                summarizeType = KruizeConstants.JSONKeys.CLUSTER;
+            // by default, db_flag will be false so the data will be fetched from cache only
+            if (fetchFromDB == null || fetchFromDB.isEmpty())
```

Comment:
based on description should be renamed 
as fetchFromCache
fetchFrom(DB or Thanos) varaible can be used when we move to Thanos server
Just a thought so you can ignore this comment .